src/config/mongoose_config_validator_toml.erl

```diff
@@ -397,16 +397,1133 @@ validate([<<"max_retry_delay">>, <<"s2s">>|Path],
+    validate_boolean(V);
+validate([<<"allow_query_users">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{allow_query_users, V}]) ->
+    validate_boolean(V);
+validate([<<"allow_user_invites">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{allow_user_invites, V}]) ->
+    validate_boolean(V);
+validate([<<"allow_visitor_nickchange">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{allow_visitor_nickchange, V}]) ->
+    validate_boolean(V);
+validate([<<"allow_visitor_status">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{allow_visitor_status, V}]) ->
+    validate_boolean(V);
+validate([<<"anonymous">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{anonymous, V}]) ->
+    validate_boolean(V);
+validate([<<"description">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{description, V}]) ->
+    validate_binary(V);
+validate([<<"logging">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{logging, V}]) ->
+    validate_boolean(V);
+validate([<<"max_users">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{max_users, V}]) ->
+    validate_positive_integer(V);
+validate([item, <<"maygetmemberlist">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [V]) ->
+    validate_non_empty_atom(V);
+validate([<<"members_by_default">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{members_by_default, V}]) ->
+    validate_boolean(V);
+validate([<<"members_only">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{members_only, V}]) ->
+    validate_boolean(V);
+validate([<<"moderated">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{moderated, V}]) ->
+    validate_boolean(V);
+validate([<<"password">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{password, V}]) ->
+    validate_string(V);
+validate([<<"password_protected">>, <<"default_room">>,
+          <<"mod_muc">>, <<"modules">>|_],
+         [{password_protected, V}]) ->
+    validate_boolean(V);
+validate([<<"persistent">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{persistent, V}]) ->
+    validate_boolean(V);
+validate([<<"public">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{public, V}]) ->
+    validate_boolean(V);
+validate([<<"public_list">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{public_list, V}]) ->
+    validate_boolean(V);
+validate([<<"subject">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{subject, V}]) ->
+    validate_string(V);
+validate([<<"subject_author">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{subject_author, V}]) ->
+    validate_string(V);
+validate([<<"title">>, <<"default_room">>, <<"mod_muc">>, <<"modules">>|_],
+         [{title, V}]) ->
+    validate_string(V);
```

Comment:
I think the option `hibernate_timeout` is missing in validation?
```suggestion
    validate_string(V);
validate([<<"hibernate_timeout">>, <<"mod_muc">>, <<"modules">> | _],
         [{hibernate_timeout, V}]) ->
    validate_non_negative_integer_or_infinity(V);
```
It is described as an option for `mod_muc`.